File: src/main/java/org/zibble/discordmessenger/util/PermissionUtil.java
package org.zibble.discordmessenger.util;

import org.zibble.discordmessenger.components.Permission;
import org.zibble.discordmessenger.components.User;

public class PermissionUtil {

    public static boolean checkPermission(User member, Permission... permissions) {
        long effectivePerms = member.getEffectivePermission();
        return isApplied(effectivePerms, Permission.ADMINISTRATOR.getRawValue())
                || isApplied(effectivePerms, Permission.getRaw(permissions));
    }

    private static boolean isApplied(long permissions, long perms) {
        return (permissions & perms) == perms;
    }

}

File: src/main/kotlin/org/zibble/discordmessenger/util/PermissionUtil.kt
package org.zibble.discordmessenger.util

import org.zibble.discordmessenger.components.entity.Permission
import org.zibble.discordmessenger.components.entity.User

object PermissionUtil {

    fun checkPermission(member: User, vararg permissions: Permission): Boolean {
        val effectivePerms = member.effectivePermission
        return (isApplied(effectivePerms, Permission.ADMINISTRATOR.raw)
                || isApplied(effectivePerms, Permission.getRaw(*permissions)))
    }

    private fun isApplied(permissions: Long, perms: Long): Boolean {
        return permissions and perms == perms
    }

}